torchdynamo/guards.py

```diff
@@ -334,34 +334,45 @@ def compile_check_fn(self, local_builder, global_builder):
         code_parts = (
             ["___guarded_code.valid"] + local_builder.code + global_builder.code
         )
+        # TODO(whc) maybe only the 'check_tensors' one is ambiguous? if so we can be less general..
```

Comment:
i'm thinking that for a bunch of the guards, just printing out their 'part' from the code (e.g. printing `guard failed: ___guarded_code.valid`) is specific enough to be useful, but, for some like odict_keys, it would be nice to actually print the mismatched keys

Reply:
Yeah.  The list might be small though.  There are keys, lengths, constants.